homeassistant/components/frontend/__init__.py

```diff
@@ -435,50 +434,30 @@ class IndexView(HomeAssistantView):
     requires_auth = False
     extra_urls = ['/states', '/states/{extra}']
 
-    def __init__(self, use_repo, js_option):
+    def __init__(self, repo_path, js_option):
         """Initialize the frontend view."""
-        from jinja2 import FileSystemLoader, Environment
-
-        self.use_repo = use_repo
-        self.templates = Environment(
-            autoescape=True,
-            loader=FileSystemLoader(
-                os.path.join(os.path.dirname(__file__), 'templates/')
-            )
-        )
+        self.repo_path = repo_path
         self.js_option = js_option
 
+    def get_template(self, latest):
```

Comment:
This is not cached. Should we?

Reply:
```python
self.tpl_cache = {}

if root not in self.tpl_cache:
    with open(os.path.join(…)) as file:
        self.tpl_cache[root] = jinja2.Template(file.read())

return self.tpl_cache[root]
```

Reply:
We should cache in non-dev mode.

I think in dev mode it's ok to never cache.

Reply:
Fixed.